### DISCUSSION TITLE
Slash command with dynamic number of arguments?

### DISCUSSION THREAD

**Discussion by damian-kos** [Rapptz/discord.py#9390] [Q&A]
So let's say that basing on user input (`test_dict` and `example_dict` in this case). I'd like to configure a slash command. But command takes arguments with their annotations. So if the user's input is 4 parameters I'd like my command to have 4 parameters as well but seems like they need to be hard-coded? 
```
from discord.ext import commands
import discord
from discord import app_commands
from typing import Optional


class Test(commands.Cog):
    def __init__(self, bot):
        self.bot = bot

    test_dict = {
        "param1": "name",
        "param2": "surname",
        "param3": "age",
        "param4": "price",
    }
    example_dict = {
        "param1": "What is your name?",
        "param2": "What is your surname?",
        "param3": "What is your age?",
        "param4": "What is your offer?",
    }

    @app_commands.command()
    @app_commands.rename(**test_dict)
    @app_commands.describe(**example_dict)
    async def person(
        self,
        interaction: discord.Interaction,
        param1: str,
        param2: str,
        param3: int,
        param4: str,
        param5: Optional[str],
    ):
        await interaction.response.send_message(
            f"Test slash")
        )
async def setup(bot):
    await bot.add_cog(Test(bot))
```

So instead of listed param1... in `person()` I wonder if there is a way to create them basing on dict in this example.

**Comment by mikeshardmind** [comment]
🤔 I mean, you could make use of the fact that you can construct a callback dynamically, but I *wouldn't*, you'd still be having just as much data that needs changing each time (discord.py needs the type information to tell discord what's valid user input for that slash arg) so instead of having the types be in the function directly, you'd still need them in yet another mapping, and then you'd need to generate a callback from this.
